src/editor/codeeditoreventbuilder.h

```diff
@@ -0,0 +1,43 @@
+#ifndef CODEEDITOREVENTBUILDER_H
+#define CODEEDITOREVENTBUILDER_H
+#include "codeeditorevents.h"
+
+
+class EventBuilder{
+private:
+    EventBuilder();
+    static Event * getEventNoModifier(QKeyEvent *e)
+    {
+        if(e->key() == Qt::Key_Enter || e->key() == Qt::Key_Return)
+            return new EventEnter();
+        return nullptr;
+    }
+    static Event * getEventShiftModifier(QKeyEvent *e)
+    {
+        if(e->key()==Qt::Key_BraceLeft)
+             return new EventBraceLeft();
+        if(e->key()==Qt::Key_BracketLeft)
+            return new EventBracketLeft();
+        if(e->key() == Qt::Key_Enter || e->key() == Qt::Key_Return)
+            return new EventShiftEnter();
+
+        return nullptr;
+    }
+    static Event * getEventControlModifier(QKeyEvent *e)
```

Comment:
static Event***** getEventControlModifier(QKeyEvent *e)

Reply:
?

Reply:
returned type is Event*
don't write * far away from Event